contracts/yield-source/ATokenYieldSource.sol

```diff
@@ -93,14 +96,17 @@ contract ATokenYieldSource is ERC20Upgradeable, IProtocolYieldSource, AssetManag
     initializer
     returns (bool)
   {
+    require(address(_aToken).isContract(), "ATokenYieldSource/aToken-not-contract-address");
     aToken = _aToken;
+
+    require(address(_lendingPoolAddressesProviderRegistry).isContract(), "ATokenYieldSource/lendingPoolRegistry-not-contract-address");
```

Comment:
Let's skip the `isContract()`.  It adds a lot of gas, and barely does more than a non-zero address check.

Reply:
(i.e. just a zero check)

Reply:
Done: https://github.com/pooltogether/aave-yield-source/pull/19/commits/eedf50e1fcccdc8f5cd1cac66bef5bbd8e80d99c